### PR TITLE
Store Natgateway enabled in InfrastructureStatus

### PR DESCRIPTION
**Pull request by elchead**
Is needed to decide on LB svc deployment if NAT is enabled for all zones

**How to categorize this PR?**

/area control-plane
/kind enhancement
/platform azure

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
The NATGateway status is now stored in the InfrastructureStatus
```

### CODE REVIEW

File: pkg/apis/azure/types_infrastructure.go [hunk 1]
```diff
@@ -167,7 +167,13 @@ type Subnet struct {
 	Zone *string
 	// Migrated is set when the network layout is migrated from NetworkLayoutSingleSubnet to NetworkLayoutMultipleSubnet.
 	// Only the subnet that was used prior to the migration should have this attribute set.
-	Migrated bool
+	Migrated       bool
+	NatGatewayName *string
```
Comment (kon-angelo) [inline pkg/apis/azure/types_infrastructure.go:171]: Exported fields should always get a description comment.

File: pkg/apis/azure/v1alpha1/types_infrastructure.go [hunk 2]
```diff
@@ -187,6 +187,8 @@ type Subnet struct {
 	// Migrated is set when the network layout is migrated from NetworkLayoutSingleSubnet to NetworkLayoutMultipleSubnet.
 	// Only the subnet that was used prior to the migration should have this attribute set.
 	Migrated bool `json:"migrated,omitempty"`
+	// NatGatewayName is enabled
```
Comment (kon-angelo) [inline pkg/apis/azure/v1alpha1/types_infrastructure.go:190]: I would be a bit more descriptive like maybe stating that if its nil then no NATGW is used.

Also you miss the `+optional` tag

File: pkg/apis/azure/types_infrastructure.go [hunk 3]
```diff
@@ -167,7 +167,13 @@ type Subnet struct {
...
+}
+
+// HasNatGateway returns if NAT is enabled
+func (s Subnet) HasNatGateway() bool {
+	return s.NatGatewayName != nil
```
Comment (kon-angelo) [inline pkg/apis/azure/types_infrastructure.go:176]: I don't see this function getting used.

Comment (elchead) [reply pkg/apis/azure/types_infrastructure.go:176]: it's used in the 2nd PR for helm chart installation

**Comment by elchead** [comment]
The tests cover assignment for multiple subnets. Still missing: test `extractNatGatewayInfoForSubnets` for multiple NATs

**Comment by elchead** [comment]
extended PR to store the NatGateway name and their IP addresses in a new field `NatGatewayStatus`

**Pull request closed without merging**